dvc/remote/local/__init__.py

```diff
@@ -566,3 +569,96 @@ def unprotect(path_info):
     @staticmethod
     def protect(path_info):
         os.chmod(path_info.path, stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH)
+
+    def append_verififcation_suffix(self, string):
+        return string + "." + self.STATUS_VERIFICATION_DIR_SUFFIX
+
+    def get_verification_dir_info(self, checksum):
+        verification_dir_info = self.checksum_to_path_info(checksum)
+        verification_dir_info.path = self.append_verififcation_suffix(
+            verification_dir_info.path
+        )
+
+        return verification_dir_info
+
+    def create_status_verification_dir(self, checksum, dir_info):
+        verification_dir_info = self.get_verification_dir_info(checksum)
+
+        self.makedirs(verification_dir_info)
+
+        for entry in dir_info:
+            entry_cache_info = self.checksum_to_path_info(
+                entry[self.PARAM_CHECKSUM]
+            )
+            relpath = entry[self.PARAM_RELPATH]
+
+            verification_entry_info = copy(entry_cache_info)
+            verification_entry_info.path = self.ospath.join(
+                verification_dir_info.path, relpath
+            )
+            self.hardlink(entry_cache_info, verification_entry_info)
+
+        self.state.save(verification_dir_info, checksum)
+
+    def _path_metadata_up_to_date(self, path):
+        mtime, size = get_mtime_and_size(path)
+        inode = get_inode(path)
+
+        entry = self.state.get_state_record_for_inode(inode)
+
+        if entry:
+            state_mtime, state_size, _, _ = entry
+            if state_mtime == mtime and state_size == size:
+                return True
+        return False
+
+    def _is_verification_dir_up_to_date(self, checksum):
+        status_verification_dir_info = self.get_verification_dir_info(checksum)
+
+        if os.path.exists(status_verification_dir_info.path):
+            return self._path_metadata_up_to_date(
+                status_verification_dir_info.path
+            )
+        return False
+
+    def _changed_dir_cache(self, checksum):
+        if self._is_verification_dir_up_to_date(checksum):
+            return False
+        return super(RemoteLOCAL, self)._changed_dir_cache(checksum)
+
+    def _save_dir(self, path_info, checksum):
+        dir_info = super(RemoteLOCAL, self)._save_dir(path_info, checksum)
+        self.create_status_verification_dir(checksum, dir_info)
```

Comment:
I don't see this method being called for `changed_cache` method though. E.g. imagine you've cache files, but you don't want to `checkout` and now you run `dvc status`, which will check for cache integrity.

Reply:
Maybe `create_status_verification_dir` should be called when loading dir cache(`load_dir_cache`) and, if needed, when `dump_dir_cache`?

Reply:
Summary of private discussion:
creation of status verification dir in load_dir_cache will be problematic, because verification dir hardlinks to cache files, which are non existent (yet) when loading dir cache upon `dvc add`.
Also updating upon `changed_cache` is probably not required, because:
- if directory changed, it will be saved as new md5.dir, and "new" verification dir will be created 
- there still exists case when one corrupts cache files, so metadata of verififcation dir changes. changed_cache will return true, but this situation is clear cache corruption, when we don't want to update cache metadata in state database, because its invalid.